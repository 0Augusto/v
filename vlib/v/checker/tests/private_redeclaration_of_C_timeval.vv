import module_with_redeclaration as mr
import time

const used = mr.used

fn main() {
	dump(time.now()) // ensure `time` is used
	//
	ts := C.timeval{}
	dump(ts)
}
