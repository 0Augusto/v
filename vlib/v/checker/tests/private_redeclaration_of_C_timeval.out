vlib/time/time.c.v:90:11: error: struct `C.timeval` was declared as private to module `module_with_redeclaration`, so it can not be used inside module `time`
   88 |         return C.GetTickCount()
   89 |     } $else {
   90 |         ts := C.timeval{}
      |                 ~~~~~~~~~
   91 |         C.gettimeofday(&ts, 0)
   92 |         return i64(ts.tv_sec * u64(1000) + (ts.tv_usec / u64(1000)))
vlib/v/checker/tests/private_redeclaration_of_C_timeval.vv:9:10: error: struct `C.timeval` was declared as private to module `module_with_redeclaration`, so it can not be used inside module `main`
    7 |     dump(time.now()) // ensure `time` is used
    8 |     //
    9 |     ts := C.timeval{}
      |             ~~~~~~~~~
   10 |     dump(ts)
   11 | }
